docs/docs/yaml-config.md

```diff
@@ -0,0 +1,153 @@
+
+## Command line interface
+
+A YAML configuration file can be passed to the command line interface with the `--config` argument. For example, say the example configuration above is saved as `example_config.yaml`. To run prediction:
+
+```console
+$ zamba predict --config example_config.yaml
+```
+
+Only some of the possible parameters can be passed directly as arguments to the command line. Those not listed in `zamba predict --help` or `zamba train --help` must be passed in a YAML file (see the [Quickstart guide](quickstart.md#getting-help) for details).
+
+## Python package
+
+The main API for zamba is the `ModelManager` class that can be accessed with:
+<!-- TODO: add link to source code><!--> 
+
+```python
+from zamba.models.manager import ModelManager
+```
+
+The `ModelManager` class is used by `zamba`’s command line interface to handle preprocessing the filenames, loading the videos, serving them to the model, and saving predictions. Therefore any functionality available to the command line interface is accessible via the `ModelManager` class.
+
+To instantiate the `ModelManager` based on a configuration file saved at `test_config.yaml`:
+```python
+>>> manager = ModelManager.from_yaml('test_config.yaml')
+>>> manager.config
+
+ModelConfig(
+  video_loader_config=VideoLoaderConfig(crop_bottom_pixels=None, i_frames=False, 
+                                        scene_threshold=None, megadetector_lite_config=None, 
+                                        video_height=224, video_width=224, 
+                                        total_frames=16, ensure_total_frames=True, 
+                                        fps=None, early_bias=False, frame_indices=None,
+                                        evenly_sample_total_frames=False, pix_fmt='rgb24'
+                                      ), 
+  train_config=None, 
+  predict_config=PredictConfig(data_directory=PosixPath('vids'), 
+                               filepaths=                         filepath
+                                          0    /home/ubuntu/vids/eleph.MP4
+                                          1  /home/ubuntu/vids/leopard.MP4
+                                          2    /home/ubuntu/vids/blank.MP4
+                                          3    /home/ubuntu/vids/chimp.MP4, 
+                                checkpoint='zamba_time_distributed.ckpt', 
+                                model_params=ModelParams(scheduler=None, scheduler_params=None),
+                                model_name='time_distributed', species=None, 
+                                gpus=1, num_workers=7, batch_size=8, 
+                                save=True, dry_run=False, proba_threshold=None,
+                                output_class_names=False, weight_download_region='us', 
+                                cache_dir=None, skip_load_validation=False)
+                              )
+```
+
```

Comment:
If we're going to mention the `ModelManager` class, let's note how you'd use it for prediction and training (`ModelManager.predict()` / `ModelManager.train()`). In other sections of the docs we use `predict_model`/`train_model` imported directly from `model_manager.py` and here we talk about using `ModelManager`, which may be a bit confusing at first. We might mention how these are related here (`ModelManager` runs `predict_model` and `train_model` under the hood).